litmus-portal/authentication/api/routes/user_router.go

```diff
@@ -0,0 +1,184 @@
+package routes
+
+import (
+	"litmus/litmus-portal/authentication/api/middleware"
+	"litmus/litmus-portal/authentication/api/presenter"
+	"litmus/litmus-portal/authentication/pkg/entities"
+	"litmus/litmus-portal/authentication/pkg/user"
+	"litmus/litmus-portal/authentication/pkg/utils"
+
+	"github.com/gin-gonic/gin"
+	log "github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+//UserRouter creates all the required routes for user authentications purposes.
+func UserRouter(router *gin.Engine, service user.Service) {
+	router.POST("/login", loginUser(service))
+	router.Use(middleware.JwtMiddleware())
+	router.POST("/update/password", updatePassword(service))
+	router.POST("/reset/password", resetPassword(service))
+	router.POST("/create", createUser(service))
+	router.POST("/update/details", updateUser(service))
+	router.GET("/users", fetchUsers(service))
+}
+
+func createUser(service user.Service) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		userRole := c.MustGet("role").(string)
+		if entities.Role(userRole) != entities.RoleAdmin {
+			c.AbortWithStatusJSON(utils.ErrorStatusCodes[utils.ErrUnauthorised], presenter.CreateErrorResponse(utils.ErrUnauthorised))
+			return
+		}
+		var userRequest entities.User
+		err := c.BindJSON(&userRequest)
+		if err != nil {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		if userRequest.Role == "" || userRequest.UserName == "" || userRequest.Password == "" || userRequest.Email == "" {
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
```

Comment:
Better to sanitize the data before creating, so that trailing spaces could be trimmed